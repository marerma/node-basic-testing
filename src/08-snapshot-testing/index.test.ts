// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

const values1 = ['one', 'two'];
const resList = {
  value: 'one',
  next: { value: 'two', next: { value: null, next: null } },
};

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    expect(generateLinkedList(values1)).toStrictEqual(resList);
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    expect(generateLinkedList(['three'])).toMatchSnapshot(
      "{ value: 'three', next: { value: null, next: null } }",
    );
  });
});
