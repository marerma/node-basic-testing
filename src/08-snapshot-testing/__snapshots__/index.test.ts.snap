// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2: { value: 'three', next: { value: null, next: null } } 1`] = `
{
  "next": {
    "next": null,
    "value": null,
  },
  "value": "three",
}
`;
